generator/src/swam/generator/ImportGenerator.scala

```diff
@@ -0,0 +1,163 @@
+package swam
+package generator
+
+import java.nio.file.{Paths, StandardOpenOption}
+
+import swam.ValType.{F32, F64, I32, I64}
+import swam.runtime.Import
+
+import scala.reflect.runtime.universe._
+import org.scalafmt.interfaces.Scalafmt
+import org.fusesource.scalate.TemplateEngine
+import cats.effect.{Blocker, Effect, IO}
+import cats.implicits._
+import cats.effect._
+import fs2.{text, _}
+
+/**
+  * @author Javier Cabrera-Arteaga on 2020-03-06
+  * Generates the scala code with the template to use as Import in the WASM execution
+  */
+class ImportGenerator[F[_]: Effect](implicit cs: ContextShift[F]) {
+
+  val f = "IO"
+  val scalafmt = Scalafmt.create(this.getClass.getClassLoader)
+  val config = Paths.get(".scalafmt.conf")
+
+  private var templateFile = getClass.getClassLoader.getResource("import_template.mustache").getFile
+  var registeredFunctions: Set[String] = Set[String]()
```

Comment:
It would be great if there were no `var` at all. The `registeredFunctions` one seems not to be used anymore anyway.